lib/dgram.js

```diff
@@ -125,6 +126,20 @@ function Socket(type, listener) {
     recvBufferSize,
     sendBufferSize
   };
+
+  if (options?.signal) {
+    const { signal } = options;
+    validateAbortSignal(signal, 'options.signal');
+    const onAborted = () => {
+      this.close();
+    };
+    if (signal.aborted) {
+      process.nextTick(onAborted);
```

Comment:
If the signal is aborted synchronously after the call the `.close` does not get deferred a microtask - is that intentional?

Reply:
i.e. - I would probably either move the `this.close()` into a nextTick or move this case out of one.

Reply:
removed `process.nextTick`